controllers/servicebinder.go

```diff
@@ -143,7 +143,11 @@ func (b *serviceBinder) unbind() (reconcile.Result, error) {
 
 	if err := b.binder.unbind(); err != nil {
 		logger.Error(err, "On unbinding related objects")
-		return requeueError(err)
+		if errors.Is(err, errApplicationNotFound) {
+			b.logger.Info("Bound application has been deleted!", "sbr.Namespace", b.binder.sbr.Namespace, "sbr.Name", b.binder.sbr.Name)
+		} else {
+			return requeueError(err)
+		}
```

Comment:
```suggestion
		return requeueError(err)
```

Reply:
@DhritiShikhar , the suggested change will break the unbind logic.   I update the commit to make the application not found error handling more clear.  

in short, we need to requeue error in general, but if the error is `application not found`, then we need to accept the error and remove finalized to unbind sbr.